mne/io/artemis123/artemis123.py

```diff
@@ -0,0 +1,263 @@
+
+    # build mne info struct
+    info = _empty_info(float(header_info['Rate Out']))
+
+    # Attempt to get time/date from fname
+    # Artemis123 files saved from the scanner observe the following
+    # naming convention 'Artemis_Data_YYYY-MM-DD-HHh-MMm_[chosen by user].bin'
+    try:
+        date = datetime.datetime.strptime(
+            op.basename(fname).split('_')[2], '%Y-%m-%d-%Hh-%Mm')
+        meas_date = calendar.timegm(date.utctimetuple())
+    except Exception:
+        meas_date = None
+
+    # build subject info
+    subject_info = {'id': header_info['Subject ID']}
+
+    # build description
+    desc = ''
+    for k in ['Purpose', 'Notes']:
+        desc += '{} : {}\n'.format(k, header_info[k])
+    desc += 'Comments : {}'.format(header_info['comments'])
+
+    info = _empty_info(float(header_info['Rate Out']))
+    info.update({'filename': fname, 'meas_date': meas_date,
+                 'description': desc, 'buffer_size_sec': 1.,
+                 'subject_info': subject_info,
+                 'proj_name': header_info['Project Name']})
+
+    # Channel Names by type
+    ref_mag_names = ['REF_001', 'REF_002', 'REF_003',
+                     'REF_004', 'REF_005', 'REF_006']
+
+    ref_grad_names = ['REF_007', 'REF_008', 'REF_009',
+                      'REF_010', 'REF_011', 'REF_012']
+
+    # load mne loc dictionary
+    loc_dict = _load_mne_locs()
+    info['chs'] = []
+    info['bads'] = []
+
+    for i, chan in enumerate(header_info['channels']):
+        # build chs struct
+        t = {'cal': float(chan['scaling']), 'ch_name': chan['name'],
+             'logno': i + 1, 'scanno': i + 1, 'range': 1.0,
+             'unit_mul': FIFF.FIFF_UNITM_NONE,
+             'coord_frame': FIFF.FIFFV_COORD_DEVICE}
+        t['loc'] = loc_dict.get(chan['name'], np.zeros(12))
+
+        if (chan['name'].startswith('MEG')):
+            t['coil_type'] = FIFF.FIFFV_COIL_ARTEMIS123_GRAD
+            t['kind'] = FIFF.FIFFV_MEG_CH
+            # TODO Shouldn't gradiometers be in T/m (Not sure if it matters)
+            t['unit'] = FIFF.FIFF_UNIT_T
+            t['unit_mul'] = FIFF.FIFF_UNITM_F
+
+        # 3 axis referance magnetometers
+        elif (chan['name'] in ref_mag_names):
+            t['coil_type'] = FIFF.FIFFV_COIL_ARTEMIS123_REF_MAG
+            t['kind'] = FIFF.FIFFV_REF_MEG_CH
+            t['unit'] = FIFF.FIFF_UNIT_T
+            t['unit_mul'] = FIFF.FIFF_UNITM_F
+
+        # reference gradiometers
+        elif (chan['name'] in ref_grad_names):
+            t['coil_type'] = FIFF.FIFFV_COIL_ARTEMIS123_REF_GRAD
+            t['kind'] = FIFF.FIFFV_REF_MEG_CH
+            # TODO Shouldn't gradiometers be in T/m (Not sure if it matters)
+            t['unit'] = FIFF.FIFF_UNIT_T
+            t['unit_mul'] = FIFF.FIFF_UNITM_F
+
+        # other reference channels are unplugged and should be ignored.
+        elif (chan['name'].startswith('REF')):
+            t['coil_type'] = FIFF.FIFFV_COIL_NONE
```

Comment:
I think we'll want these to still be configured with proper units. For example when concatenating two files, one that had a "bad"/off reference channel, it will be more difficult to do so if we say that one doesn't have a coil type, but the other does. It *should* suffice in the code just to mark it as bad.

@agramfort do you agree? I haven't had to explicitly think about an unplugged channel before, but I guess you can have equivalently flat channels in EEG and those just get marked bad, not set to channel type misc ...